data/modules/TradeShips/Debug.lua

```diff
@@ -0,0 +1,333 @@
+				{ name = "Message", key = "msg",                             string = true }
+			},{
+				callbacks = {
+					onClick =	function(row)
+						status = Core.ships[row.ship] and Core.ships[row.ship].status
+						if status and status ~= "hyperspace" and status ~= "hyperspace_out" then
+							Game.systemView:SetSelectedObject(Engine.GetEnumValue("ProjectableTypes", "OBJECT"),
+								Engine.GetEnumValue("ProjectableBases", "SHIP"), row.ship)
+						end
+					end,
+					isSelected = function(row)
+						return ship_selected and obj.ref == row.ship
+					end
+			}})
+	end
+end)
+
+infosize = ui.getCursorScreenPos().y
+local obj = Game.systemView:GetSelectedObject()
+if obj.type ~= Engine.GetEnumValue("ProjectableTypes", "NONE") and Core.ships[obj.ref] then
+	local ship = obj.ref
+	local trader = Core.ships[ship]
+	if ui.collapsingHeader("Info:", {"DefaultOpen"}) then
+		property("Trader: ", ship:GetShipType() .. " " .. ship:GetLabel())
+		local status = trader.status
+		if status == "docked" then
+			status = status .. " (" .. trader.starport:GetLabel() .. ")"
+		elseif status == "inbound" then
+			local d,u = ui.Format.Distance(ship:DistanceTo(trader.starport))
+			status = status .. " (" .. trader.starport:GetLabel() .. " - " .. d .. " " ..  u .. ")"
+		end
+		property("Status: ", status)
+		if trader.fnc then
+			property("Task: ", trader.fnc .. " in " .. ui.Format.Duration(trader.delay - Game.time))
+		end
+		property("Fuel: ", string.format("%.4f", ship.fuelMassLeft) .. "/" .. ShipDef[ship.shipId].fuelTankMass .. " t")
+	end
+	if ui.collapsingHeader("Internals:", {"DefaultOpen"}) then
+		local equipItems = {}
+		local total_mass = 0
+		local equips = { "cargo", "misc", "hyperspace", "laser" }
+		for _,t in ipairs(equips) do
+			for _,et in pairs(e[t]) do
+				local count = ship:CountEquip(et)
+				if count > 0 then
+					local all_mass = count * et.capabilities.mass
+					table.insert(equipItems, {
+							name = et:GetName(),
+							eq_type = t,
+							count = count,
+							mass = et.capabilities.mass,
+							all_mass = all_mass
+						})
+					total_mass = total_mass + all_mass
+				end
+			end
+		end
+		local capacity = ShipDef[ship.shipId].capacity
+		drawArrayTable("tradeships_traderequipment", equipItems, ipairs,
+			{
+				{ name = "Name",        key = "name",                          string = true },
+				{ name = "Type",        key = "eq_type",                       string = true },
+				{ name = "Units",       key = "count"                                        },
+				{ name = "Unit's mass", key = "mass",     fnc = format("%dt")                },
+				{ name = "Total",       key = "all_mass", fnc = format("%dt")                }
+			},
+			{ totals = {
+					{ name = "Total:",    all_mass = total_mass            },
+					{ name = "Capacity:", all_mass = capacity              },
+					{ name = "Free:",     all_mass = capacity - total_mass },
+			}})
```

Comment:
Same thing here, this is doubly-indented. There are several instances of similar cases in this file (and possibly others); I'm only going to point this one out for brevity's sake.

Reply:
Done!